### DISCUSSION TITLE
Config help with regex manager matching for using `releaseTimestamp`

### DISCUSSION THREAD

**Discussion by pathob** [renovatebot/renovate#26181] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest-with-local-path

### If you're self-hosting Renovate, select which platform you are using.

github.com

### What is your question?

I'm trying to experiment with the `releaseTimestamp` attribute that is basically available in many datasources (such as Maven), but not exposed yet for use in templates.

For this reason, I added `releaseTimestamp` to `allowedFields` and run the patched version locally, but I have a few issues and I couldn't solve them spending all day yesterday...

Here is my example (also hosted at https://github.com/pathob/renovate-release-timestamps)

```
# renovate.json

{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:recommended"],
  "prHourlyLimit": 0,
  "prConcurrentLimit": 0,
  "rebaseWhen": "conflicted",
  "fetchChangeLogs": "off",
  "customManagers": [
    {
      "customType": "regex",
      "fileMatch": ["(^|/)release\\.yaml$"],
      "matchStrings": [
        "artifact:\\n(?<indentation>[ ]+?)type: (?<datasource>[^\\s]*?)\\s*name: (?<depName>[^\\s]*?)\\s*version: (?<currentValue>[^\\s]*?)\\s*releaseTimestamp: (?:[^\\s]*?)\\s*registryUrl: (?<registryUrl>[^\\s]*?)\\s"
      ],
      "autoReplaceStringTemplate": "artifact\n{{{indentation}}}type: {{{datasource}}}\n{{{indentation}}}name: {{{depName}}}\n{{{indentation}}}version: {{{newValue}}}\n{{{indentation}}}releaseTimestamp: {{{releaseTimestamp}}}\n{{{indentation}}}registryUrl: {{{registryUrl}}}"
    }
  ]
}
```

```
# release.yaml
# (just an example, we don't use it like this in case someone wonders)

components:
  junit:
    artifact:
      type: maven
      name: org.junit.jupiter:junit-jupiter-api
      version: 5.9.1
      releaseTimestamp: 2022-09-20T00:00:00.000Z
      registryUrl: https://repo1.maven.org/maven2/
      packaging: jar
  slf4j:
    artifact:
      type: maven
      name: org.slf4j:slf4j-api
      version: 1.7.35
      releaseTimestamp: 2022-01-25T00:00:00.000Z
      registryUrl: https://repo1.maven.org/maven2/
      packaging: jar
```

The first question is: Is my approach correct? The main problem that I currently have is that it seems that renovate is replacing too much in my `release.yaml` file. Output:

```
DEBUG: Starting search at index 256 (repository=pathob/renovate-release-timestamps, packageFile=release.yaml, branch=renovate/org.slf4j-slf4j-api-2.x)
       "depName": "org.slf4j:slf4j-api"
DEBUG: Found match at index 256 (repository=pathob/renovate-release-timestamps, packageFile=release.yaml, branch=renovate/org.slf4j-slf4j-api-2.x)
       "depName": "org.slf4j:slf4j-api"
DEBUG: Extracted release.yaml after autoreplace has fewer deps than expected. (repository=pathob/renovate-release-timestamps, branch=renovate/org.slf4j-slf4j-api-2.x)
 WARN: Error updating branch: update failure (repository=pathob/renovate-release-timestamps, branch=renovate/org.slf4j-slf4j-api-2.x)
```

I checked my regex with regex101 and cannot find any issue with it...

![image](https://github.com/renovatebot/renovate/assets/4580157/aae8afb7-70ce-4a06-ad88-bdee32650e4e)

I also have the problem, which makes it quite hard to test changes to `autoReplaceStringTemplate`, that Renovate does not update existing branches, because of `autoReplaceStringTemplate` being `true` in the `doAutoReplace` method and I don't know whether this can be manipulated...

**Comment by rarkins** [comment]
You've asked "is my approach correct" but I don't know what it is you're trying to do, other than "I'm trying to experiment". Can you summarize your business/application requirements, avoiding Renovate terminology as much as possible. e.g. you have a file in a repo, and it contains x,y and you need it to be updated based on z, etc.

**Reply by pathob** [reply]
Okay, I just want to update the versions in the provided release.yaml file together with the corresponding release timestamps.

I had it working without the `releaseTimestamp`, only with the `fileMatch` option, so without the `autoReplaceStringTemplate` option. Since it didn't work with `releaseTimestamp` anymore, I thought `autoReplaceStringTemplate` is the way to go.

**Reply by pathob** [reply]
Hi @rarkins may I please ask for help again? Thank you!

**Reply by rarkins** [reply]
Sorry, I'm not sure of any answers

**Reply by pathob** [reply]
Okay... now I'm confused...

@viceice maybe? I mean setting another template value besides the version should be possible, or not?!
